src/main/java/org/sagebionetworks/bridge/fitbit/worker/BridgeFitBitWorkerProcessor.java

```diff
@@ -178,6 +187,15 @@ void processStudy(String dateString, Study study) {
                     }
                 } catch (Exception ex) {
                     LOG.error("Error getting next user: " + ex.getMessage(), ex);
+
+                    // The Iterator is a paginated iterator that calls Bridge for each user. If for some reason, it
+                    // keeps throwing exceptions (for example, Bridge is down), this could retry infinitely. Cap the
+                    // number of errors, and if we hit that threshold, break out of the loop and propagate the
+                    // exception up the call stack.
+                    numErrors++;
+                    if (numErrors >= userErrorLimit) {
+                        throw new WorkerException("User error limit reached, aborting for study " + studyId);
+                    }
```

Comment:
Could possibly use the generic LimitExceededException here.

Reply:
I'd prefer to avoid mixing Bridge Server exceptions and Bridge Worker exceptions.

Reply:
It's a generic exception in the Java standard libraries.